airflow/models.py

```diff
@@ -1061,6 +1060,7 @@ def are_dependencies_met(
 
         task = self.task
 
+        logging.info("Checkpoint A")
```

Comment:
Absolutely, everything after my initial commit in this pr is actually profiling stuff (ie. WIP). A lot of time is spent in are_dependencies_met as it is iterating over *all* tasks now every time now.